Firestore/core/test/firebase/firestore/model/field_value_test.cc

```diff
@@ -42,6 +42,109 @@ TEST(FieldValue, BooleanType) {
   EXPECT_TRUE(false_value < true_value);
 }
 
+TEST(FieldValue, NumberType) {
+  const FieldValue nan_value = FieldValue::NanValue();
+  const FieldValue integer_value = FieldValue::IntegerValue(10L);
+  const FieldValue double_value = FieldValue::DoubleValue(10.1);
+  EXPECT_EQ(Type::Double, nan_value.type());
+  EXPECT_EQ(Type::Long, integer_value.type());
+  EXPECT_EQ(Type::Double, double_value.type());
+  EXPECT_TRUE(nan_value < integer_value);
```

Comment:
EXPECT_LT?

Reply:
I think that's a safe assumption to make. Also it will give better error messages, but it's not that big of deal one way or the other.